### PR TITLE
defult caseblock date format to full datetime

### PR DESCRIPTION
**Pull request by snopoke**
@TylerSheffels 
cc @czue

### CODE REVIEW

**Comment by czue** [comment]
dear god, yes. i wonder how many tests are going to fail...

**Comment by snopoke** [comment]
Probably a few

**Comment by czue** [comment]
image

that didn't take log

File: corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def testBasicRebuild(self):
         case.xform_ids = [case.xform_ids[0], case.xform_ids[2], case.xform_ids[1]]
         case.rebuild()
         self.assertEqual(case.p1, 'p1-1') # original
-        self.assertEqual(case.p2, 'p2-1') # updated (back!)
+        self.assertEqual(case.p2, 'p2-2') # updated (back!)
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py]: E261 at least two spaces before inline comment

**Comment by snopoke** [comment]
:white_check_mark:

File: corehq/ex-submodules/casexml/apps/case/mock.py [hunk 2]
```diff
@@ -213,7 +213,7 @@ def __init__(self,
                     '_text': case_id
                 }
 
-    def as_xml(self, format_datetime=date_to_xml_string):
+    def as_xml(self, format_datetime=None):
```
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/case/mock.py]: This is the actual change

File: corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py (hunk 1 above)
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py]: This has always just been wrong

Comment (czue) [inline corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py]: i'm confused by this - I think the test was intending to test that if you flipped the action order then `p2-1` would win out over `p2-2`. I think what may be happening now is that the rebuild process is also sorting the forms which now have granular enough dates that flipping the action order gets reverted during the sorting that happens in rebuild. so this test is correct but no longer testing the thing it was originally intended to test.

can we just set all the action dates to be the same here? or override the datetime format?

Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/case/tests/test_rebuild.py]: Oh, I was assuming it was testing that rebuilding with out of order actions but perhaps that's what the next test is doing. I'll set the dates to the same value which should put it back into the previous state.

**Comment by snopoke** [comment]
:white_check_mark:

**Pull request merged**